js/ui/appDisplay.js

```diff
@@ -2807,7 +2810,50 @@ class HackAppIcon extends AppIcon {
                 DownstreamEndlessIconState.HACK_DEACTIVATED;
         });
 
+        this._hackPulseId = global.settings.connect('changed::hack-icon-pulse', () => {
+            this._pulse = global.settings.get_boolean('hack-icon-pulse');
+            if (this._pulseWaitId) {
+                Mainloop.source_remove(this._pulseWaitId);
+                this._pulseWaitId = 0;
+            }
+            this.pulse();
+        });
+
         this.canDrop = false;
+
+        this.pulse();
+    }
+
+    pulse() {
+        if (!this._pulse)
+            return;
+
+        const params = { time: 0.1, transition: 'easeOutQuad' };
+        this._tweenIcon({...params, scale_x: 1.1, scale_y: 1.1})
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 0.9, scale_y: 0.9}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 1.1, scale_y: 1.1}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 0.9, scale_y: 0.9}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 1, scale_y: 1}))
+            .then(() => {
+                this._pulseWaitId = Mainloop.timeout_add(1000, () => {
+                    this._pulseWaitId = 0;
+                    this.pulse();
+                    return GLib.SOURCE_REMOVE;
+                });
+            });
+    }
+
+    _tweenIcon(tweenParams) {
+        // Getting the icon without the label
+        const actor = this.icon._iconBin.first_child;
```

Comment:
This is pretty much a no-go... can't you use `this.icon.icon` here?

Reply:
Yes sure, I was looking into the `BaseIcon` constructor and I didn't found the `icon` param at first so I was using the path that I found using looking glass.